### ISSUE TITLE
Regardless of deactivating load generator, we have communication between microservices

### ISSUE THREAD

**Issue by Aliasghar-Rahmanian** [GoogleCloudPlatform/microservices-demo#353]
I want not to have any auto-generated load for the application. To do so, I removed load generator service, however, I still see bunch of interactions between microservices by tshark!
I want see interactions between microservice if and only if I send a request to the frontend service using CURL. Is there any reason we have such interaction between microservices?

**Comment by ahmetb** [comment]
You should not be still seeing requests between services. There’s a chance loadgenerator has not terminated yet. 

Feel free to inspect the source code and see where the traffic occurs by also checking logs. 

There’s not much we can do, I don’t think we do requests in the background.

**Comment by askmeegs** [comment]
@ahmetb when a grpc client is initialized, wouldn't that produce traffic between services, on 
 startup (eg. from frontend to productcatalog?)  - even when the loadgen isn't running?

**Comment by ahmetb** [comment]
Probably need to inspect the source code, but my guess is no. While writing the app we made sure frontend requests go to the backends at request-time (for example, the productcatalog is not preloaded before a request arrives, and it's re-fetched on every request).

This is assuming you define traffic as RPCs; not bytes on the wire.

**Comment by Aliasghar-Rahmanian** [comment]
> You should not be still seeing requests between services. There’s a chance loadgenerator has not terminated yet.
> 
> Feel free to inspect the source code and see where the traffic occurs by also checking logs.
> 
> There’s not much we can do, I don’t think we do requests in the background.

I used release/kubernetes_manifest.yaml file to deploy application on cluster of virtual box VMs. Before doing that, I removed load generator service in the yaml file and thus there must not be any load for the application, is that right? I used "tshark -i any" inside the frontend container and see packet transmissions between this service and other services (i.e. currency-service, shipping-service, cart-service, product-catalog-service, checkout-service, and adservice). I also inspected other services by tshark and saw the same issue.

**Comment by ahmetb** [comment]
> I used "tshark -i any" inside the frontend container and see packet transmissions between this service and other services

Yes. That's why I said 

> This is assuming you define traffic as RPCs; not bytes on the wire.

To validate frontend can connect to services, it has to establish a TCP connection (and possibly do tls negotiation).

**Comment by askmeegs** [comment]
Hi @ahmetb, did you mean to reopen this? Happy to investigate this bug further.

**Comment by ahmetb** [comment]
Yeah sorry didn’t mean to close, but it’s likely there’s not much we can do to stop all tcp traffic.

**Comment by Aliasghar-Rahmanian** [comment]
I tried to inspect what's going on. Now, I believe it may be because of the liveliness & readiness probe that causes seeing bunch of packet transmissions regardless of incoming demands. Hence, I increased periodSeconds in the yaml files as much as I could  in order to remove their side effects. It helped! and I did not see packet transmission between microservices, however, bunch of local to local packet transmissions are there instead. I think it would be the gRPC health checking mechanism that makes this happen.
